build.gradle

```diff
@@ -81,10 +81,13 @@ task bundledJar(type: Jar) {
 	}
 
 	from {
-		configurations.compile.collect { it.isDirectory() ? it : zipTree(it) }
+		configurations.compileClasspath.collect { it.isDirectory() ? it : zipTree(it) }
 	}
 
-	archiveName = "${project.name}-bundled.jar"
+	archiveFileName = "${project.name}-bundled.jar"
+
+	exclude('META-INF/LICENSE.txt')
```

Comment:
I believe that these changes are in violation of the Apache License (v2, Section 4, Pragraphs a and d). If we remove licenses and notices from our distribution, then we should take care to reproduce them otherwise, e.g. print them as a special command or so.

Reply:
I agree - however, the original state was in violation of that as well since both the license and notice files were overwritten with each other.
I'd rather not introduce a special command for something like that, especially since it would mean having those files not managed through gradle's dependency management anymore.

I think if we want to  fix this, the way to go would be using the [Shadow Plugin](https://github.com/johnrengelman/shadow). However, the plugin itself still produces deprecation warnings as of release 5.2.0, see [this issue](https://github.com/johnrengelman/shadow/issues/530).
Since this is not really a time-critical change, I'd suggest keeping this PR open until version 6.0 of the Shadow Plugin is released

Reply:
I would prefer a solution that does not add another dependency to the project. Having a low number of dependencies has a several of advantages, one of which is not having to wait for someone else to fix their (and now our) problems. Is there no way in Gradle to rename a file if a file with the same name already exists?

Reply:
Since the build file is essentially a groovy script, that's doable.. I pushed a possible fix without using any other plugins, just numbering license and notice files. Feels a bit hacky tbh, but works..

Reply:
Given the status quo, I think it is an okay solution.